client-sniffer/src/test/java/org/elasticsearch/client/sniff/SnifferBuilderTests.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.sniff;
+
+import com.carrotsearch.randomizedtesting.generators.RandomInts;
+import com.carrotsearch.randomizedtesting.generators.RandomPicks;
+import org.apache.http.HttpHost;
+import org.apache.lucene.util.LuceneTestCase;
+import org.elasticsearch.client.RestClient;
+
+import java.util.Arrays;
+import java.util.logging.LogManager;
+
+public class SnifferBuilderTests extends LuceneTestCase {
```

Comment:
ES's core tests, the client's tests, the sniffer's tests, and any libs that we pull out so the client can share them. Like if we pull TimeValue into some `:common:lib` or something. I expect it'll be much more interesting as we go to higher level java clients. Because I wasn't sure we needed it I didn't push for it.

I like having the sniffer be separate because it adds a dependency and folks that are paranoid about dependencies in their software might not want it.